source/lac/scalapack.cc

```diff
@@ -673,19 +673,77 @@ void ScaLAPACKMatrix<NumberType>::compute_cholesky_factorization()
 
 
 
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::compute_LU_factorization(std::vector<int> &ipiv)
+{
+  if (grid->mpi_process_is_active)
+    {
+      int info = 0;
+      NumberType *A_loc = &this->values[0];
+
+      const int iarow = indxg2p_(&submatrix_row, &row_block_size, &(grid->this_process_row), &first_process_row, &(grid->n_process_rows));
+      const int mp = numroc_(&n_rows, &row_block_size, &(grid->this_process_row), &iarow, &(grid->n_process_rows));
+      ipiv.resize(mp+row_block_size);
+
+      pgetrf(&n_rows,&n_columns,A_loc,&submatrix_row,&submatrix_column,descriptor,ipiv.data(),&info);
+      AssertThrow (info==0, LAPACKSupport::ExcErrorCode("pgetrf", info));
+    }
+  state = LAPACKSupport::State::lu;
+}
+
+
+
 template <typename NumberType>
 void ScaLAPACKMatrix<NumberType>::invert()
 {
-  if (state == LAPACKSupport::matrix)
```

Comment:
please keep `if`, the point is that users should be able to do 
```
compute_cholesky()
invert()
```
or call directly
```
invert()
```

Reply:
so just call your `compute_lu_factorization()` if it's not symmetric and is in the matrix state.

Reply:
Hopefully, I addressed that appropriately in the latest commit :)